### PR TITLE
Feature/persistent registry

### PR DESCRIPTION
**Pull request by andre-merzky**
As the name says: the PR adds optional persistence to the ZMQ registry.

### CODE REVIEW

File: src/radical/utils/zmq/registry.py [hunk 1]
```diff
@@ -11,16 +13,23 @@
...
+    The `ru.zmq.Registry` is a ZMQ service which provides a hirarchical
+    persistent data store.
+    '''
 
     # --------------------------------------------------------------------------
     #
     def __init__(self, url=None,
-                       uid=None) -> None:
+                       uid=None,
+                       persistent='False') -> None:
```
Comment (mtitov) [inline src/radical/utils/zmq/registry.py:25]: should be boolean type, right? -> `persistent=False`

File: src/radical/utils/zmq/registry.py [hunk 2]
```diff
@@ -11,16 +13,23 @@
 # ------------------------------------------------------------------------------
 #
 class Registry(Server):
-
+    '''
+    The `ru.zmq.Registry` is a ZMQ service which provides a hirarchical
```
Comment (mtitov) [inline src/radical/utils/zmq/registry.py:17]: typo `hirarchical` -> `hierarchical`

Comment (andre-merzky) [reply src/radical/utils/zmq/registry.py:17]: Thanks!

File: src/radical/utils/zmq/registry.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/utils/zmq/registry.py:25]: Thanks, fixed (and type hints added)

**Pull request merged**